gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/auth/AbstractDelegationTokenBinding.java

```diff
@@ -0,0 +1,266 @@
+        requireNonNull(createTokenIdentifier(renewerText),
+                       "Token identifier");
+
+    Token<AbstractGCPTokenIdentifier> token =
+        new Token<>(tokenIdentifier, secretManager);
+    token.setKind(getKind());
+    token.setService(service);
+    logger.atFine().log("Created token %s with token identifier %s",
+        token, tokenIdentifier);
+    return token;
+  }
+
+  /**
+   * Create a token identifier with all the information needed
+   * to be included in a delegation token.
+   * This is where session credentials need to be extracted, etc.
+   * This will only be called if a new DT is needed, that is: the
+   * filesystem has been deployed unbound.
+   *
+   * If {@link #createDelegationToken}
+   * is overridden, this method can be replaced with a stub.
+   *
+   * @return the token data to include in the token identifier.
+   *
+   * @throws IOException failure creating the token data.
+   */
+  public abstract AbstractGCPTokenIdentifier createTokenIdentifier(Text renewer)
+      throws IOException;
+
+
+  /**
+   * Create a token identifier with all the information needed
+   * to be included in a delegation token.
+   * This is where session credentials need to be extracted, etc.
+   * This will only be called if a new DT is needed, that is: the
+   * filesystem has been deployed unbound.
+   *
+   * If {@link #createDelegationToken}
+   * is overridden, this method can be replaced with a stub.
+   *
+   * @return the token data to include in the token identifier.
+   *
+   * @throws IOException failure creating the token data.
+   */
+  public abstract AbstractGCPTokenIdentifier createTokenIdentifier()
+      throws IOException;
+
+
+  /**
+   * Create a new subclass of {@link AbstractGCPTokenIdentifier}.
+   * This is used in the secret manager.
+   * @return an empty identifier.
+   */
+  public abstract AbstractGCPTokenIdentifier createEmptyIdentifier();
+
+
+  /**
+   * Verify that a token identifier is of a specific class.
+   * This will reject subclasses (i.e. it is stricter than
+   * {@code instanceof}, then cast it to that type.
+   * @param identifier identifier to validate
+   * @param expectedClass class of the expected token identifier.
+   * @throws DelegationTokenIOException If the wrong class was found.
+   */
+  protected <T extends AbstractGCPTokenIdentifier> T convertTokenIdentifier(
+      final AbstractGCPTokenIdentifier identifier,
+      final Class<T> expectedClass) throws DelegationTokenIOException {
+    if (!identifier.getClass().equals(expectedClass)) {
+      throw DelegationTokenIOException.wrongTokenType(expectedClass, identifier);
+    }
+    return (T) identifier;
+  }
+
+  /**
+   * The secret manager always uses the same secret; the
+   * factory for new identifiers is that of the token manager.
+   */
+  protected class TokenSecretManager extends SecretManager<AbstractGCPTokenIdentifier> {
+
+    private final byte[] pwd = "not-a-pass".getBytes(StandardCharsets.UTF_8);
```

Comment:
Could you add a comment here explaining this arbitrary value, just for clarification?

Reply:
This is not actually used, but it provided to satisfy the SecretManager requirement of the associated API. The s3a connector does something similar.

Reply:
@steveloughran I think it'd be good to add a brief comment here, just so someone reading the code knows not to be alarmed by it. Would you have a suggestion?